app/screens/edit_profile/edit_profile.tsx

```diff
@@ -157,14 +160,29 @@ const EditProfile = ({
         setError(undefined);
         setUpdating(true);
         try {
-            const newUserInfo: Partial<UserProfile> = {
-                email: userInfo.email.trim(),
-                first_name: userInfo.firstName.trim(),
-                last_name: userInfo.lastName.trim(),
-                nickname: userInfo.nickname.trim(),
-                position: userInfo.position.trim(),
-                username: userInfo.username.trim(),
-            };
+            // Build update object with only changed and unlocked fields
+            const newUserInfo: Partial<UserProfile> = {};
+
+            // Only include fields that have changed and are not locked by SAML
+            if (userInfo.email.trim() !== currentUser.email && !currentUser.authService) {
+                newUserInfo.email = userInfo.email.trim();
+            }
+            if (userInfo.firstName.trim() !== currentUser.firstName && !lockedFirstName) {
+                newUserInfo.first_name = userInfo.firstName.trim();
+            }
+            if (userInfo.lastName.trim() !== currentUser.lastName && !lockedLastName) {
+                newUserInfo.last_name = userInfo.lastName.trim();
+            }
+            if (userInfo.nickname.trim() !== currentUser.nickname && !lockedNickname) {
+                newUserInfo.nickname = userInfo.nickname.trim();
+            }
+            if (userInfo.position.trim() !== currentUser.position && !lockedPosition) {
+                newUserInfo.position = userInfo.position.trim();
+            }
+            if (userInfo.username.trim() !== currentUser.username && !currentUser.authService) {
+                newUserInfo.username = userInfo.username.trim();
+            }
```

Comment:
0/5 I wonder if we can make this more concise with something like...
```
const baseAttributes: Array<keyof UserProfile> = ['email', 'firstName', 'lastName', 'nickname', 'position'];
baseAttributes.forEach((key) => {
    const trimmedValue = userInfo[key].trim();
    if (trimmedValue !== currentUser[key] && !locked[key]) {
        newUserInfo[key] = trimmedValue;
    }
}
```
You would also have to update how you are passing the "locked" variables, and include the lockedEmail and lockedUsername to be derived from authService. But it would leave this piece of code cleaner.

But again... 0/5, maybe this is good enough.